### PR TITLE
feat(cluster): Allow appending RDB to existing store

### LINKED ISSUE #2840: Add support loading multiple rdb files
DFLY LOAD [NOFLUSH] filename

### PR DESCRIPTION
**Pull request by chakaz**
The goal of this PR is to support the loadoing of multiple RDB files into a single server, like when migrating from a Valkey cluster to Dragonfly with a different number of nodes.

It makes the following changes:

1. Removes `DEBUG LOAD`, as we already have `DFLY LOAD`
2. Adds `APPEND` option to `DFLY LOAD` (i.e. `DFLY LOAD <filename> APPEND`) that loads an RDB _without_ first flushing the data store, overriding existing keys
3. Does not load keys belonging to unowned slots, if in cluster mode

Fixes #2840

### CODE REVIEW

File: src/server/debugcmd.cc [hunk 1]
```diff
@@ -500,7 +495,19 @@ void DebugCmd::Reload(CmdArgList args) {
...
   string last_save_file = sf_.GetLastSaveInfo().file_name;
-  Load(last_save_file);
+
+  sf_.FlushAll(cntx_);
+
+  if (auto fut_ec = sf_.Load(last_save_file, ServerFamily::LoadExistingKeys::kFail); fut_ec) {
+    GenericError ec = fut_ec->Get();
+    if (ec) {
+      string msg = ec.Format();
+      LOG(WARNING) << "Could not load file " << msg;
```
Comment (BorysTheDev) [inline src/server/debugcmd.cc:505]: I think it's error

File: src/server/dflycmd.cc [hunk 2]
```diff
@@ -163,11 +163,33 @@ void DflyCmd::Run(CmdArgList args, ConnectionContext* cntx) {
...
-  if (sub_cmd == "LOAD" && args.size() == 2) {
-    DebugCmd debug_cmd{sf_, cntx};
-    debug_cmd.Load(ArgS(args, 1));
-    return;
+  if (sub_cmd == "LOAD") {
+    return Load(args, cntx);
+  }
+
+  if (sub_cmd == "HELP") {
+    string_view help_arr[] = {
```
Comment (BorysTheDev) [inline src/server/dflycmd.cc:171]: It strange for me that help returns array, please consider to use R"()" string literal

File: src/server/server_family.cc [hunk 3]
```diff
@@ -935,13 +934,40 @@ struct AggregateLoadResult {
...
-std::optional<fb2::Future<GenericError>> ServerFamily::Load(const std::string& load_path) {
+std::optional<fb2::Future<GenericError>> ServerFamily::Load(string_view load_path,
+                                                            LoadExistingKeys existing_keys) {
+  fs::path path(load_path);
+
+  if (load_path.empty()) {
+    fs::path dir_path(GetFlag(FLAGS_dir));
+    string filename = GetFlag(FLAGS_dbfilename);
+    dir_path.append(filename);
+    path = dir_path;
```
Comment (BorysTheDev) [inline src/server/server_family.cc:959]: I think it should work like this "dir_path / filename;"

File: src/server/server_family.h [hunk 4]
```diff
@@ -16,6 +16,7 @@
 #include "server/dflycmd.h"
 #include "server/engine_shard_set.h"
 #include "server/namespaces.h"
+#include "server/rdb_load.h"
```
Comment (BorysTheDev) [inline src/server/server_family.h:19]: why do you need this include

**Review by BorysTheDev — DISMISSED** [review]

File: src/server/dflycmd.cc (hunk 2 above)
Comment (kostasrim) [reply src/server/dflycmd.cc:171]: It's a usual thing to return an array for commands in valkey

File: src/server/debugcmd.cc (hunk 1 above)
Comment (chakaz) [reply src/server/debugcmd.cc:505]: I moved this code from below
We shouldn't get alerts for failing debug commands IMO

File: src/server/dflycmd.cc (hunk 2 above)
Comment (chakaz) [reply src/server/dflycmd.cc:171]: Indeed, as a human user it's easier to see errors like this than "line1\nline2" etc

File: src/server/server_family.cc (hunk 3 above)
Comment (chakaz) [reply src/server/server_family.cc:959]: This code is moved from debugcmd, I didn't touch it :)
But using `fs::path` is better than appending strings for various reasons

File: src/server/server_family.h (hunk 4 above)
Comment (chakaz) [reply src/server/server_family.h:19]: Leftovers, will remove

File: src/server/server_family.cc (hunk 3 above)
Comment (BorysTheDev) [reply src/server/server_family.cc:960]: I suggest to use fs::path syntax instead of append you can use / operator for path concatenation

**Review by BorysTheDev — APPROVED** [review]

**Pull request merged**